src/games/strategy/engine/framework/EngineVersionProperties.java

```diff
@@ -71,79 +58,42 @@ private static EngineVersionProperties contactServerForEngineVersionProperties(f
     // so run a couple requests in parallel, starting with delays to try and get a response quickly
     final AtomicReference<EngineVersionProperties> ref = new AtomicReference<>();
     final CountDownLatch latch = new CountDownLatch(1);
-    final Runnable r = new Runnable() {
-      @Override
-      public void run() {
-        for (int i = 0; i < 5; i++) {
-          spawnRequest(engineversionPropsURL, ref, latch);
-          try {
-            latch.await(2, TimeUnit.SECONDS);
-          } catch (final InterruptedException e) {
-          }
-          if (ref.get() != null) {
-            break;
-          }
-        }
-        // we have spawned a bunch of requests
-        try {
-          latch.await(15, TimeUnit.SECONDS);
-        } catch (final InterruptedException e) {
-        }
+    for (int i = 0; i < 5; i++) {
+      new Thread(() -> {
+        ref.set(new EngineVersionProperties(engineversionPropsURL));
+        latch.countDown();
+      }).start();
+      try {
+        latch.await(2, TimeUnit.SECONDS);
+      } catch (final InterruptedException e) {
       }
-
-      private void spawnRequest(final URL engineversionPropsURL, final AtomicReference<EngineVersionProperties> ref,
-          final CountDownLatch latch) {
-        final Thread t1 = new Thread(new Runnable() {
-          @Override
-          public void run() {
-            ref.set(new EngineVersionProperties(engineversionPropsURL));
-            latch.countDown();
-          }
-        });
-        t1.start();
+      if (ref.get() != null) {
+        break;
       }
-    };
-    r.run();
```

Comment:
It's stupid to create an anonymous Runnable class just to call the run method...